In [2]:
#test query for smoking/diabetes

import sqlite3

conn=sqlite3.connect("/Users/dc/clinical.sqlite")
c = conn.cursor()

c.execute("select distinct(user_id) from problems where problem_name like '%smoke%';")
print c.fetchall()

#
c.execute("select distinct(user_id) from problems where problem_name like '%smoking%';")
print c.fetchall()

#synonym expansion, words next to smoking for all 4 tables
#tobacco (from table problems)
#affecting (from table problems)
#cessation (from table problems)
#OF (from table problems)
#synonym expaneion, words 2 away 
# HX  (from table problems)
# HISTORY  (from table problems)
# PREGNANCY  (from table problems)
# COUNSELING  (from table problems)
#------------------------------------
# 
# sqlite> select * from medications where medication_name like '%smoking%';
#212|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|993518|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|rxnorm|extracted|56fab5216465d212000e32dc
#212|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|65084-414-42|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|ndc|extracted|56fab5216465d212000e32dc
#212|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|84764|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|medispandrugdescriptorid|extracted|56fab5216465d212000e32dc
#212|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|62100002107430|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|medispangenericproductidentifier|extracted|56fab5216465d212000e32dc
#212|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|84764|Bupropion Hcl Er (Smoking Det) 150 Mg Or Tb12|medfile(medispan)|extracted|56fab5216465d212000e32dc
# Bupropion words 3 away
#no results from 2 other tables
#sqlite> select * from allergies where allergy_name like '%smoking%';
#sqlite> select * from immunizations where immunization_name like '%smoking%';

#problem: synonym expansion doesn't pick up nicotine patch!
c.execute("select distinct(user_id) from medications where medication_name like '%nicotine%';")
print c.fetchall()


c.execute("select distinct(user_id) from problems where problem_name like '%copd%';")
print c.fetchall()

#tables allergies and immunizaitons had no results as seen above. 

[(u'212',), (u'2079',), (u'1923',), (u'74',), (u'2044',), (u'397',), (u'931',)]
[(u'1952',), (u'2142',), (u'359',), (u'1781',)]
[(u'212',), (u'2475',), (u'2118',), (u'312',), (u'1914',), (u'2212',), (u'112',)]
[(u'1593',), (u'596',), (u'1609',)]


<h4>There are different codes for smoking</h4>

<p>
1952|SMOKING CESSATION COUNSELING|V65.42|SMOKING CESSATION COUNSELING|icd9cm|extracted|56fb5d8811cb1f120046015d
1952|SMOKING CESSATION COUNSELING|Z71.6|SMOKING CESSATION COUNSELING|icd10cm|extracted|56fb5d8811cb1f120046015d
2142|Tobacco smoking affecting pregnancy|427189007|Tobacco smoking affecting pregnancy|snomedct|extracted|56ec124944de480100c5ad63
2142|Tobacco smoking affecting pregnancy|649.00|Tobacco smoking affecting pregnancy|icd9cm|extracted|56ec124944de480100c5ad63
2142|Tobacco smoking affecting pregnancy|O99.330|Tobacco smoking affecting pregnancy|icd10cm|extracted|56ec124944de480100c5ad63
359|Tobacco smoking affecting pregnancy in first trimester|427189007|Tobacco smoking affecting pregnancy in first trimester|snomedct|extracted|56dca59db1b5490100418931
359|Tobacco smoking affecting pregnancy in first trimester|649.03|Tobacco smoking affecting pregnancy in first trimester|icd9cm|extracted|56dca59db1b5490100418931
359|Tobacco smoking affecting pregnancy in first trimester|O99.331|Tobacco smoking affecting pregnancy in first trimester|icd10cm|extracted|56dca59db1b5490100418931
1781|HISTORY OF SMOKING|V15.82|HISTORY OF SMOKING|icd9cm|extracted|56d51a5f5219770100de807a
1781|HISTORY OF SMOKING|Z87.891|HISTORY OF SMOKING|icd10cm|extracted|56d51a5f5219770100de807a
1781|HISTORY OF SMOKING|392521001|Medical History|snomedct|inferred|56d51a5f5219770100de807a
1781|HX OF SMOKING|V15.82|HX OF SMOKING|icd9cm|extracted|56d51a495219770100de7d64
1781|HX OF SMOKING|Z87.891|HX OF SMOKING|icd10cm|extracted|56d51a495219770100de7d64
</p>

<h4>Tobacco keyword</h4>
Tobacco is not always synonymous with smoking. ie tobacco chewing

sqlite> select * from problems where problem_name like '%tobacco%';
46|TOBACCO USER|110483000|TOBACCO USER|snomedct|extracted|56f764a2f7cea312007ef66c
46|TOBACCO USER|305.1|TOBACCO USER|icd9cm|extracted|56f764a2f7cea312007ef66c
46|TOBACCO USER|Z72.0|TOBACCO USER|icd10cm|extracted|56f764a2f7cea312007ef66c
9|Tobacco abuse|89765005|Tobacco abuse|snomedct|extracted|56f7162d67d7a901000bc9c9
9|Tobacco abuse|305.1|Tobacco abuse|icd9cm|extracted|56f7162d67d7a901000bc9c9
9|Tobacco abuse|Z72.0|Tobacco abuse|icd10cm|extracted|56f7162d67d7a901000bc9c9
2475|Tobacco use|110483000|Tobacco use|snomedct|extracted|56f1f118a6f66a0100e0d36d
2475|Tobacco use|305.1|Tobacco use|icd9cm|extracted|56f1f118a6f66a0100e0d36d
2475|Tobacco use|Z72.0|Tobacco use|icd10cm|extracted|56f1f118a6f66a0100e0d36d
74|TOBACCO SMOKER|77176002|TOBACCO SMOKER|snomedct|extracted|56f0a04265ff3a1200e0c343
74|TOBACCO SMOKER|305.1|TOBACCO SMOKER|icd9cm|extracted|56f0a04265ff3a1200e0c343
74|TOBACCO SMOKER|F17.200|TOBACCO SMOKER|icd10cm|extracted|56f0a04265ff3a1200e0c343
2142|Tobacco smoking affecting pregnancy|427189007|Tobacco smoking affecting pregnancy|snomedct|extracted|56ec124944de480100c5ad63
2142|Tobacco smoking affecting pregnancy|649.00|Tobacco smoking affecting pregnancy|icd9cm|extracted|56ec124944de480100c5ad63
2142|Tobacco smoking affecting pregnancy|O99.330|Tobacco smoking affecting pregnancy|icd10cm|extracted|56ec124944de480100c5ad63
2142|Tobacco use|110483000|Tobacco use|snomedct|extracted|56ec124944de480100c5acca
2142|Tobacco use|305.1|Tobacco use|icd9cm|extracted|56ec124944de480100c5acca
2142|Tobacco use|Z72.0|Tobacco use|icd10cm|extracted|56ec124944de480100c5acca
774|Tobacco abuse|89765005|Tobacco abuse|snomedct|extracted|56e62d06804f6e0100f2fa73
774|Tobacco abuse|305.1|Tobacco abuse|icd9cm|extracted|56e62d06804f6e0100f2fa73
774|Tobacco abuse|Z72.0|Tobacco abuse|icd10cm|extracted|56e62d06804f6e0100f2fa73
1440|Tobacco abuse|89765005|Tobacco abuse|snomedct|extracted|56e2bfa706e7280100f31a13
1440|Tobacco abuse|305.1|Tobacco abuse|icd9cm|extracted|56e2bfa706e7280100f31a13
1440|Tobacco abuse|Z72.0|Tobacco abuse|icd10cm|extracted|56e2bfa706e7280100f31a13
1794|Tobacco use disorder|89765005|Tobacco use disorder|snomedct|extracted|56e26eb33da9a60100843272
1794|Tobacco use disorder|305.1|Tobacco use disorder|icd9cm|extracted|56e26eb33da9a60100843272
1794|Tobacco use disorder|Z72.0|Tobacco use disorder|icd10cm|extracted|56e26eb33da9a60100843272
2153|Tobacco use disorder|89765005|Tobacco use disorder|snomedct|extracted|56e08f0cb81e7c01004d578a
2153|Tobacco use disorder|305.1|Tobacco use disorder|icd9cm|extracted|56e08f0cb81e7c01004d578a
2153|Tobacco use disorder|Z72.0|Tobacco use disorder|icd10cm|extracted|56e08f0cb81e7c01004d578a
397|TOBACCO SMOKER|77176002|TOBACCO SMOKER|snomedct|extracted|56de4a9ec6f3f301009cea66
397|TOBACCO SMOKER|305.1|TOBACCO SMOKER|icd9cm|extracted|56de4a9ec6f3f301009cea66
397|TOBACCO SMOKER|F17.200|TOBACCO SMOKER|icd10cm|extracted|56de4a9ec6f3f301009cea66
248|Tobacco dependence|89765005|Tobacco dependence|snomedct|extracted|56de461cb81e7c01004c2efc
248|Tobacco dependence|305.1|Tobacco dependence|icd9cm|extracted|56de461cb81e7c01004c2efc
248|Tobacco dependence|F17.200|Tobacco dependence|icd10cm|extracted|56de461cb81e7c01004c2efc
2354|Tobacco use|110483000|Tobacco use|snomedct|extracted|56dce39cb81e7c01004b78aa
2354|Tobacco use|305.1|Tobacco use|icd9cm|extracted|56dce39cb81e7c01004b78aa
2354|Tobacco use|Z72.0|Tobacco use|icd10cm|extracted|56dce39cb81e7c01004b78aa
359|Tobacco smoking affecting pregnancy in first trimester|427189007|Tobacco smoking affecting pregnancy in first trimester|snomedct|extracted|56dca59db1b5490100418931
359|Tobacco smoking affecting pregnancy in first trimester|649.03|Tobacco smoking affecting pregnancy in first trimester|icd9cm|extracted|56dca59db1b5490100418931
359|Tobacco smoking affecting pregnancy in first trimester|O99.331|Tobacco smoking affecting pregnancy in first trimester|icd10cm|extracted|56dca59db1b5490100418931
268|Tobacco abuse, in remission|191889006|Tobacco abuse, in remission|snomedct|extracted|56dbc15ac278dd01006a62bf
268|Tobacco abuse, in remission|V15.82|Tobacco abuse, in remission|icd9cm|extracted|56dbc15ac278dd01006a62bf
268|Tobacco abuse, in remission|F17.201|Tobacco abuse, in remission|icd10cm|extracted|56dbc15ac278dd01006a62bf

<h4>System benchmarking</h4>

In [24]:
#well these results are awful lets try another stemmer
from stemming.porter2 import stem

c.execute("select distinct(problem_name) from problems;")
list_Problems = c.fetchall()
print len(list_Problems)


list_stemmedProblems =[]

for problem in list_Problems:
    stemProblem = stem(problem[0].lower())
    if stemProblem !=problem[0].lower():
        print "problem:", problem[0].lower(), " stemmed:", stemProblem
        list_stemmedProblems.append(stemProblem)

print 'num stemmed:', len(list_stemmedProblems)

2813
problem: disease  stemmed: diseas
problem: history of gestational diabetes  stemmed: history of gestational diabet
problem: major depressive disorder, recurrent episode, in partial remission  stemmed: major depressive disorder, recurrent episode, in partial remiss
problem: encounter for screening colonoscopy  stemmed: encounter for screening colonoscopi
problem: chronic kidney disease  stemmed: chronic kidney diseas
problem: ascus with positive high risk hpv - cin 1 on colpo - repap/annual in 6 months  stemmed: ascus with positive high risk hpv - cin 1 on colpo - repap/annual in 6 month
problem: acute sinusitis, unspecified  stemmed: acute sinusitis, unspecifi
problem: obesity  stemmed: obes
problem: acute maxillary sinusitis  stemmed: acute maxillary sinus
problem: migraine  stemmed: migrain
problem: cellulitis and abscess of toe, unspecified  stemmed: cellulitis and abscess of toe, unspecifi
problem: depressive disorder, not elsewhere classified  stemmed: depressive disorder, no

In [22]:
import sqlite3

conn=sqlite3.connect("/Users/dc/clinical.sqlite")
c = conn.cursor()
from nltk.stem.snowball import SnowballStemmer
from nltk.stem.lancaster import LancasterStemmer
from nltk.stem import WordNetLemmatizer


lan_stem = LancasterStemmer()
snow_stem = SnowballStemmer('english')
word_lemma = WordNetLemmatizer()


def ls(wordstring):
    word_list = wordstring.split()
    stemmed=[]
    for word in word_list:
        stemmed.append(lan_stem.stem(word))
    return stemmed

def ss(wordstring):
    word_list = wordstring.split()
    stemmed=[]
    for word in word_list:
        stemmed.append(snow_stem.stem(word))
    return stemmed

def wl(wordstring):
    word_list = wordstring.split()
    stemmed=[]
    for word in word_list:
        stemmed.append(word_lemm.lemmatize(word))
    return stemmed

c.execute("select distinct(problem_name) from problems;")
list_Problems = c.fetchall()
print len(list_Problems)

# stem phrases has no effect
num=0
for problem in list_Problems:
    if (num%100) ==0:
        print num,": problem",problem, "snowballStem:", snow_stem.stem(problem[0]), " lancasterStem:", lan_stem.stem(problem[0]), " lemmatizer:", word_lemma.lemmatize(problem[0])
    num +=1
print 'end test'

        

    
print '----------------------------------------------'
#stem each word in a phrase
num=0
for problem in list_Problems:
    if num%100 ==0:
        print num,": problem:", problem, "snowballStem:", snow_stem.stem(problem[0]), "lancasterStemmer:", lan_stem.stem(problem[0]), "lemmatizer:", word_lemma.lemmatize(problem[0]) 
    num+=1
    
    
        
        
        

2813
0 : problem (u'Disease',) snowballStem: diseas  lancasterStem: diseas  lemmatizer: Disease
100 : problem (u'BACK PAIN',) snowballStem: back pain  lancasterStem: back pain  lemmatizer: BACK PAIN
200 : problem (u'Oligomenorrhea (finding)',) snowballStem: oligomenorrhea (finding)  lancasterStem: oligomenorrhea (finding)  lemmatizer: Oligomenorrhea (finding)
300 : problem (u'Ehlers-Danlos syndrome',) snowballStem: ehlers-danlos syndrom  lancasterStem: ehlers-danlos syndrome  lemmatizer: Ehlers-Danlos syndrome
400 : problem (u'Sleep disturbance, unspecified',) snowballStem: sleep disturbance, unspecifi  lancasterStem: sleep disturbance, unspecified  lemmatizer: Sleep disturbance, unspecified
500 : problem (u'Anxiety problem',) snowballStem: anxiety problem  lancasterStem: anxiety problem  lemmatizer: Anxiety problem
600 : problem (u'MACROMASTIA',) snowballStem: macromastia  lancasterStem: macromast  lemmatizer: MACROMASTIA
700 : problem (u'SVT (supraventricular tachycardia) (HC code)',